app/views/viewlet-manager.js

```diff
@@ -266,13 +268,21 @@ YUI.add('juju-viewlet-manager', function(Y) {
         view.render(this.getAttrs());
         viewContainer.append(view.get('container'));
         this.databindingBind(model, view);
-      } else {
+      } else if (typeof view.render === 'function') {
+        // We only pre-render if it's a YUI view.
         var result = view.render(model, this.getAttrs());
         if (result && typeof result === 'string') {
           view.container = Y.Node.create(result);
         }
         viewContainer.append(view.container);
         this.databindingBind(model, view);
+      } else {
+        // If this is a React component we need to create a container so that it
+        // will play nice with the Y.View's and then we need to store that
+        // container with the React component constructor.
+        var wrapperNode = Y.Node.create('<div/>');
+        viewContainer.append(wrapperNode);
+        view.wrapperElement = wrapperNode.getDOMNode();
```

Comment:
not in the the viewlet manager takes care of cleaning all these elements up when its destroyed - for places like the machine view this funkyness wouldn't be necessary.